Content.Client/Overlays/ShowSecurityIconsSystem.cs

```diff
@@ -0,0 +1,76 @@
+using Content.Shared.Access.Components;
+using Content.Shared.Body.Components;
+using Content.Shared.Inventory;
+using Content.Shared.Overlays;
+using Content.Shared.PDA;
+using Content.Shared.StatusIcon;
+using Content.Shared.StatusIcon.Components;
+using Robust.Shared.Prototypes;
+
+namespace Content.Client.Overlays
+{
+    public sealed class ShowSecurityIconsSystem : ComponentAddedOverlaySystemBase<ShowSecurityIconsComponent>
+    {
+        [Dependency] private readonly IPrototypeManager _prototypeMan = default!;
+        [Dependency] private readonly InventorySystem _inventorySystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<BodyComponent, GetStatusIconsEvent>(OnGetStatusIconsEvent);
+        }
+
+        private void OnGetStatusIconsEvent(EntityUid uid, BodyComponent _, ref GetStatusIconsEvent @event)
+        {
+            if (!IsActive || @event.InContainer)
+            {
+                return;
+            }
+
+            var healthIcons = DecideSecurityIcon(uid);
+
+            @event.StatusIcons.AddRange(healthIcons);
+        }
+
+        private IReadOnlyList<StatusIconPrototype> DecideSecurityIcon(EntityUid uid)
+        {
+            var result = new List<StatusIconPrototype>();
+
+            var iconToGet = "NoId";
+            if (_inventorySystem.TryGetSlotEntity(uid, "id", out var idUid))
+            {
+                // PDA
+                if (TryComp(idUid, out PdaComponent? pda))
+                {
+                    iconToGet = pda.ContainedId?.JobTitle ?? string.Empty;
+                }
+                // ID Card
+                else if (TryComp(idUid, out IdCardComponent? id))
+                {
+                    iconToGet = id.JobTitle ?? string.Empty;
+                }
+
+                iconToGet = iconToGet.Replace(" ", "");
```

Comment:
> * Each ID card has a `StatusIconPrototype` data field. Use this for the overlay instead of inferring an icon from the job name.

This would need to be configurable for Agent IDs before it can be a solution.

> * Each `JobPrototype` points to some `StatusIconPrototype`

Using the JobPrototype is unacceptable to me for gameplay as it it a meta-component. 

> * Look at uses of `IdCardSystem.TryChangeJobTitle()`. If they set the job based on some JobPrototype (i.e., not via a HOP console), they should probably also set the icon to match the job prototype's icon.

50/50 split. Two uses set it from a job and the other two take it from a user's input field.

> * Eventually, HOP console should be allowed to configure the icon similar to how it can configure the job name, though that should be a separate PR.
I think this would be best when combined with the Agent ID. I sure hope they both share the same FE code.

Reply:
> Using the JobPrototype is unacceptable to me for gameplay as it it a meta-component.

Maybe I wasn't clear enough, but this is just meant to be used to set the default icon when a job is assigned, in exactly the same way as its used to assign the default job title.